code-challenges.test.js

```diff
@@ -14,38 +14,110 @@
+describe("removeShuffle", () => {
+  it("removes the first item from the array and returns the remaining content shuffled", () => {
+      const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+      // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
+    expect(removeShuffle(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+      const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+      // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+    expect(removeShuffle(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+  })
+})
+// ReferenceError: removeShuffle is not defined
 
 // b) Create the function that makes the test pass.
+// pseudocode
+// Create a function called removeShuffle
+// Input will be an array
+// Remove the first item in the array using .shift()
+// Shuffle the array (didn't know how to do this initially, so had to look it up and saw how to use the .sort() and Math.random() method for this problem)
+// Return the array
+// Output will be an array with the first item removed, and the rest of the content is shuffled
+
+const removeShuffle = (array) => {
+  array.shift()
+  array.sort(() => Math.random() - 0.5)
+  return array
+}
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
+
+
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
-
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+describe("tallyBalance", () => {
+  it("returns the end tally", () => {
+      const votes1 = { upVotes: 13, downVotes: 2 }
+      // Expected output: 11
+    expect(tallyBalance(votes1)).toEqual(11)
+      const votes2 = { upVotes: 2, downVotes: 33 }
+      // Expected output: -31
+    expect(tallyBalance(votes2)).toEqual(-31)
+  })
+})
+// ReferenceError: tallyBalance is not defined
 
 // b) Create the function that makes the test pass.
+// pseudocode
+// Create a function called tallyBalance
+// Input is an object
+// Need to access the values in the object
+// Combine the upVotes with downVotes (downVotes are considered negative while upVotes are positive) so the equation would be upVotes - downVotes
+// Output is a number that contains the end tally balance
+
+const tallyBalance = (object) => {
+  return object.upVotes - object.downVotes   // not sure if there is a more dynamic way to do this? will the values always have 'upVotes' and 'downVotes' as the keys? if they will, then this will work. if not, I'm not completely sure how to access an object's values without knowing the name of the key.
```

Comment:
This is the best approach as far as a further refactor, I don't think it needs one.